Here is the hadoop map/red configuration (conf/mapred-site.xml) that fails


         <property>
                <name>mapred.child.java.opts</name>
                <value>-Xmx300m -XX:+DoEscapeAnalysis -XX:+UseCompressedOops -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC</value>
        </property>



Here is the hadoop map/red configuration that succeeds


         <property>
                <name>mapred.child.java.opts</name>
                <value>-Xmx300m -XX:+DoEscapeAnalysis -XX:+UseCompressedOops</value>
        </property>



Here is the exception from the pig script.


Backend error message
---------------------
org.apache.pig.backend.executionengine.ExecException: ERROR 2081: Unable to set up the load function.
        at org.apache.pig.backend.executionengine.PigSlice.init(PigSlice.java:89)
        at org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.SliceWrapper.makeReader(SliceWrapper.java:144)
        at org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.PigInputFormat.getRecordReader(PigInputFormat.java:282)
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:338)
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:307)
        at org.apache.hadoop.mapred.Child.main(Child.java:170)
Caused by: java.lang.RuntimeException: could not instantiate 'PigStorage' with arguments '[,]'
        at org.apache.pig.impl.PigContext.instantiateFuncFromSpec(PigContext.java:519)
        at org.apache.pig.backend.executionengine.PigSlice.init(PigSlice.java:85)
        ... 5 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
        at org.apache.pig.impl.PigContext.instantiateFuncFromSpec(PigContext.java:487)
        ... 6 more
Caused by: java.lang.RuntimeException: Couldn't find heap
        at org.apache.pig.impl.util.SpillableMemoryManager.<init>(SpillableMemoryManager.java:95)
        at org.apache.pig.data.BagFactory.<init>(BagFactory.java:106)
        at org.apache.pig.data.DefaultBagFactory.<init>(DefaultBagFactory.java:71)
        at org.apache.pig.data.BagFactory.getInstance(BagFactory.java:76)
        at org.apache.pig.builtin.Utf8StorageConverter.<init>(Utf8StorageConverter.java:49)
        at org.apache.pig.builtin.PigStorage.<init>(PigStorage.java:69)
        at org.apache.pig.builtin.PigStorage.<init>(PigStorage.java:79)
        ... 11 more

